application/electron/src/controllers/files.parsers/file.parser.text.ts

```diff
@@ -57,34 +58,13 @@ export default class FileParser extends AFileParser {
 
     public isSupported(file: string): Promise<boolean> {
         return new Promise((resolve, reject) => {
-            fs.open(file, 'r', (openError: NodeJS.ErrnoException | null, fd: number) => {
-                if (openError) {
-                    return reject(openError);
-                }
-                const buffer: Buffer = Buffer.alloc(ft.minimumBytes);
-                fs.read(
-                    fd,
-                    buffer,
-                    0,
-                    ft.minimumBytes,
-                    0,
-                    (readError: NodeJS.ErrnoException | null, read: number, buf: Buffer) => {
-                        if (readError) {
-                            return reject(readError);
-                        }
-                        const type: ft.FileTypeResult | undefined = ft(buf);
-                        if (type === undefined) {
-                            resolve(false);
-                        } else if (
-                            type.mime.indexOf('text') !== -1 ||
-                            type.mime.indexOf('application') !== -1
-                        ) {
-                            resolve(true);
-                        }
-                        resolve(false);
-                    },
-                );
-            });
+            readExactlyByFile(file, fs.statSync(file).size)
```

Comment:
Do we need to read the whole file? What if it will be a 10Gb file?

Reply:
Seems like the first 50 Byte suffice to decide if it's a binary file or not. Changed the code accordingly